src/relay/backend/graph_runtime_codegen.cc

```diff
@@ -371,10 +371,28 @@ class GraphRuntimeCodegen : public backend::MemoizedExprTranslator<std::vector<G
 
       // Step into the functions that are handled by external codegen to
       // collect metadata.
+      auto codegen = func->GetAttr<String>(attr::kCompiler);
+      CHECK(codegen.defined()) << "No external codegen is set";
+      std::string codegen_name = codegen.value();
       const auto name_node = func->GetAttr<String>(tvm::attr::kGlobalSymbol);
-      std::string symobl = std::string(name_node.value());
-      ConstantUpdater const_visit(symobl, &params_);
-      const_visit(func);
+      std::string symbol = std::string(name_node.value());
+      std::string const_update_name = "relay.ext." + codegen_name + ".constant_updater";
```

Comment:
Could you elaborate on this? I'm not sure what's meant my moving it to the updater.

Reply:
I meant that we could check the extern updaters in the constant updater itself, i.e. if there is one we use that one, otherwise we use the generic one. Then, we don't have to do have this in both graph_runtime_codegen.cc and compiler.cc.

Reply:
Do you think it might be cleaner to introduce a new function in backend/utils called something like UpdateConstants which replicates the logic that currently lives directly in graph runtime/vm? That way we can keep the generic ConstantUpdater clean.

Reply:
I think we probably don't need to though. As you can see, currently it is just one/two lines of code.

Reply:
I'm not exactly sure what you mean - these two lines specifically get the symbol from the name_node and the name of the constant updater global func. You need the symbol to initialise the ConstantUpdater, and the ConstantUpdater doesn't require global func name (because it's the default fallback option).

Reply:
I am saying that we probably don't want to duplicate the code in both compiler.cc and build_module.cc. Instead, we can move it into ConstantUpdater so that difference places can share the same code. I am not saying these two lines are not needed.

Reply:
Agreed, but there's about 20 shared lines in this change between the two, not just those two. That's why I suggested creating a shared utility function rather than moving it all into ConstantUpdater.

Reply:
I am fine with that as well